dynamodb/dynamo_lock_table.go

```diff
@@ -204,3 +215,53 @@ type TableDoesNotExist struct {
 func (err TableDoesNotExist) Error() string {
 	return fmt.Sprintf("Table %s does not exist in DynamoDB! Original error from AWS: %v", err.TableName, err.Underlying)
 }
+
+// Encrypt the TFState Lock table - If Necessary
+func UpdateLockTableSetSSEncryptionOnIfNecessary(tableName string, client *dynamodb.DynamoDB, terragruntOptions *options.TerragruntOptions) error {
+	tableSSEncrypted, err := LockTableCheckSSEncryptionIsOn(tableName, client)
+	if err != nil {
+		return err
+	}
+
+	if !tableSSEncrypted {
+		tableCreateDeleteSemaphore.Acquire()
+		defer tableCreateDeleteSemaphore.Release()
```

Comment:
Do we need this? I know there are concurrency limits on creating/deleting tables... Not sure about enabling/deleting encryption?

Reply:
@brikis98 I couldn't find good enough information on that - and I decided to put it just in case. My guess is - it's not needed. However, do we have a way to test it?

Reply:
Basically - this is updating a table. I believe the concurrency rules might be in place.

Reply:
It's not documented anywhere obvious. I just found it when we started having test failures and the errors mentioned concurrency issues due to having many tests creating/updating tests concurrently. I suppose leaving it in is OK.

Reply:
Super - leaving it in then.